crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -1,12 +1,31 @@
-use crate::system::BoxedSystem;
+use std::borrow::Cow;
+
+use crate::{
+    system::{BoxedSystem, CombinatorSystem, Combine, IntoSystem, System},
+    world::World,
+};
 
 pub type BoxedCondition = BoxedSystem<(), bool>;
 
 /// A system that determines if one or more scheduled systems should run.
 ///
 /// Implemented for functions and closures that convert into [`System<In=(), Out=bool>`](crate::system::System)
 /// with [read-only](crate::system::ReadOnlySystemParam) parameters.
-pub trait Condition<Params>: sealed::Condition<Params> {}
+pub trait Condition<Params>: sealed::Condition<Params> {
+    fn and_then<P, C: Condition<P>>(self, and_then: C) -> AndThen<Self::System, C::System> {
```

Comment:
Doc strings and doc tests here :)

I really like that these are methods though!